packages/glimmer-syntax/lib/parser.ts

```diff
@@ -38,16 +38,60 @@ export function preprocess(html, options?) {
 
 const entityParser = new EntityParser(namedCharRefs);
 
-export function Parser(source, options) {
-  this.options = options || {};
-  this.elementStack = [];
-  this.tokenizer = new EventedTokenizer(this, entityParser);
+export class Parser {
+  private elementStack: Object[] = [];
+  private options: Object;
+  private source: string[];
+  public currentAttribute = null;
+  public currentNode = null;
+  public tokenizer: EventedTokenizer = new EventedTokenizer(this, entityParser);
+
+  constructor(source: any, options: Object) {
+    this.options = options || {};
+    if (typeof source === 'string') {
+      this.source = source.split(/(?:\r\n?|\n)/g);
+    }
```

Comment:
@chancancode Updated everything except this. Haven't been able to figure out how to answer your question yet. Any help would be appreciated.